Qt6 build fails due to "ignoring return value" warning

Building kiwix-desktop against qt6 currently fails due to `QMAKE_CXXFLAGS += -Werror`:

```
src/kiwixapp.cpp: In member function ‘QString KiwixApp::parseStyleFromFile(QString)’:
src/kiwixapp.cpp:532:14: error: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Werror=unused-result]
  532 |     file.open(QFile::ReadOnly);
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/QtCore:91,
                 from /usr/include/qt6/QtNetwork/QtNetworkDepends:3,
                 from /usr/include/qt6/QtNetwork/QtNetwork:3,
                 from src/opdsrequestmanager.h:5,
                 from src/contentmanager.h:7,
                 from src/kiwixapp.h:5,
                 from src/kiwixapp.cpp:1:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
cc1plus: all warnings being treated as errors
make: *** [Makefile:2349: kiwixapp.o] Error 1
```

See https://github.com/kiwix/kiwix-desktop/blob/main/src/kiwixapp.cpp#L561C1-L568C2, @juuz0

--- a/src/choiceitem.cpp
+++ b/src/choiceitem.cpp
@@ -11,7 +11,7 @@ ChoiceItem::ChoiceItem(QString key, QString value, QWidget *parent) :
     m_value(value)
 {
     ui->setupUi(this);
-    this->setStyleSheet(KiwixApp::instance()->parseStyleFromFile(":/css/choiceBox.css"));
+    this->setStyleSheet(getFileContent(":/css/choiceBox.css"));
     ui->itemLabel->setText(key);
     ui->itemLabel->setToolTip(key);
     connect(ui->closeButton, &QPushButton::clicked, [=](){

--- a/src/contentmanagerside.cpp
+++ b/src/contentmanagerside.cpp
@@ -14,7 +14,7 @@ ContentManagerSide::ContentManagerSide(QWidget *parent) :
 {
     setFocusPolicy(Qt::FocusPolicy::StrongFocus);
     mp_ui->setupUi(this);
-    this->setStyleSheet(KiwixApp::instance()->parseStyleFromFile(":/css/contentmanagerside.css"));
+    this->setStyleSheet(getFileContent(":/css/contentmanagerside.css"));
 
     mp_ui->buttonGroup->setId(mp_ui->allFileButton, CatalogButtonId::ALL);
     mp_ui->buttonGroup->setId(mp_ui->localFileButton, CatalogButtonId::LOCAL);

--- a/src/contentmanagerview.cpp
+++ b/src/contentmanagerview.cpp
@@ -10,7 +10,7 @@ ContentManagerView::ContentManagerView(QWidget *parent)
 {
     mp_ui->setupUi(this);
     mp_ui->m_view->setSortingEnabled(true);
-    mp_ui->m_view->setStyleSheet(KiwixApp::instance()->parseStyleFromFile(":/css/_contentManager.css"));
+    mp_ui->m_view->setStyleSheet(getFileContent(":/css/_contentManager.css"));
     mp_ui->m_view->setContextMenuPolicy(Qt::CustomContextMenu);
     auto managerDelegate = new ContentManagerDelegate();
     mp_ui->m_view->setItemDelegate(managerDelegate);

--- a/src/kiwixapp.cpp
+++ b/src/kiwixapp.cpp
@@ -85,7 +85,7 @@ void KiwixApp::init()
 
     setApplicationName("Kiwix");
     setDesktopFileName("org.kiwix.desktop.desktop");
-    setStyleSheet(parseStyleFromFile(":/css/style.css"));
+    setStyleSheet(getFileContent(":/css/style.css"));
 
     createActions();
     mp_mainWindow = new MainWindow;
@@ -558,13 +558,16 @@ void KiwixApp::printVersions(std::ostream& out) {
   zim::printVersions(out);
 }
 
-QString KiwixApp::parseStyleFromFile(QString filePath)
+QString getFileContent(QString filePath)
 {
     QFile file(filePath);
-    file.open(QFile::ReadOnly);
-    QString styleSheet = QString(file.readAll());
+    if (!(file.open(QFile::ReadOnly))) {
+      qWarning("Error opening %s", qPrintable(filePath));
+      return QString();
+    }
+    QString content= QString(file.readAll());
     file.close();
-    return styleSheet;
+    return content;
 }
 
 void KiwixApp::saveListOfOpenTabs()

--- a/src/kiwixapp.h
+++ b/src/kiwixapp.h
@@ -99,7 +99,6 @@ public:
     QString getText(const QString &key) { return m_translation.getText(key); };
     void setMonitorDir(const QString &dir);
     bool isCurrentArticleBookmarked();
-    QString parseStyleFromFile(QString filePath);
     void saveListOfOpenTabs();
     void saveWindowState();
     void saveVoiceName(const QString& langName, const QString& voiceName);
@@ -147,6 +146,7 @@ private:
     void loadAndInstallTranslations(QTranslator& translator, const QString& filename, const QString& directory);
 };
 
+QString getFileContent(QString filePath);
 QString gt(const QString &key);
 #define _STR(...) # __VA_ARGS__
 #define STR(X) _STR(X)

--- a/src/kiwixchoicebox.cpp
+++ b/src/kiwixchoicebox.cpp
@@ -18,7 +18,7 @@ KiwixChoiceBox::KiwixChoiceBox(QWidget *parent) :
     ui(new Ui::kiwixchoicebox)
 {
     ui->setupUi(this);
-    auto styleSheet = KiwixApp::instance()->parseStyleFromFile(":/css/choiceBox.css");
+    auto styleSheet = getFileContent(":/css/choiceBox.css");
     this->setStyleSheet(styleSheet);
     ui->clearButton->setText(gt("clear"));
     ui->clearButton->setToolTip(gt("clear-filter"));

--- a/src/kiwixmessagebox.cpp
+++ b/src/kiwixmessagebox.cpp
@@ -10,7 +10,7 @@ KiwixMessageBox::KiwixMessageBox(QString confirmTitle, QString confirmText, bool
 {
     ui->setupUi(this);
     setWindowFlag(Qt::FramelessWindowHint, true);
-    setStyleSheet(KiwixApp::instance()->parseStyleFromFile(":/css/messageBox.css"));
+    setStyleSheet(getFileContent(":/css/messageBox.css"));
     connect(ui->yesButton, &QPushButton::clicked, [=]() {
         emit yesClicked();
         m_result = YesClicked;

--- a/src/kprofile.cpp
+++ b/src/kprofile.cpp
@@ -20,9 +20,7 @@ QWebEngineScript getScript(QString filename,
     script.setInjectionPoint(point);
     script.setWorldId(QWebEngineScript::UserWorld);
 
-    QFile scriptFile(filename);
-    scriptFile.open(QIODevice::ReadOnly);
-    script.setSourceCode(scriptFile.readAll());
+    script.setSourceCode(getFileContent(filename));
     return script;
 }
 

--- a/src/localkiwixserver.cpp
+++ b/src/localkiwixserver.cpp
@@ -13,12 +13,7 @@ LocalKiwixServer::LocalKiwixServer(QWidget *parent) :
     setWindowFlag(Qt::FramelessWindowHint, true);
     ui->setupUi(this);
 
-    QFile styleFile(":/css/localServer.css");
-    styleFile.open(QIODevice::ReadOnly);
-    auto byteContent = styleFile.readAll();
-    styleFile.close();
-    QString style(byteContent);
-    setStyleSheet(style);
+    setStyleSheet(getFileContent(":/css/localServer.css"));
 
     mp_server = KiwixApp::instance()->getLocalServer();
     m_port = KiwixApp::instance()->getSettingsManager()->getKiwixServerPort();

--- a/src/searchbar.cpp
+++ b/src/searchbar.cpp
@@ -85,7 +85,7 @@ SearchBarLineEdit::SearchBarLineEdit(QWidget *parent) :
     m_suggestionView->setItemDelegate(new SuggestionListDelegate(this));
     m_suggestionView->header()->setStretchLastSection(true);
     m_suggestionView->setRootIsDecorated(false);
-    m_suggestionView->setStyleSheet(KiwixApp::instance()->parseStyleFromFile(":/css/popup.css"));
+    m_suggestionView->setStyleSheet(getFileContent(":/css/popup.css"));
 
     const int contentHeight = HeaderSectionCSS::lineHeight;
     m_suggestionView->setIconSize(QSize(contentHeight, contentHeight));

--- a/src/settingsview.cpp
+++ b/src/settingsview.cpp
@@ -35,7 +35,7 @@ SettingsView::SettingsView(QWidget *parent)
 {
     SettingsManager *settingsMgr = KiwixApp::instance()->getSettingsManager();
     ui->setupUi(this);
-    ui->widget->setStyleSheet(KiwixApp::instance()->parseStyleFromFile(":/css/_settingsManager.css"));
+    ui->widget->setStyleSheet(getFileContent(":/css/_settingsManager.css"));
     connect(ui->zoomPercentSpinBox, QOverload<int>::of(&QSpinBox::valueChanged), this, &SettingsView::setZoom);
     connect(ui->moveToTrashToggle, &QCheckBox::clicked, this, &SettingsView::setMoveToTrash);
     connect(ui->reopenTabToggle, &QCheckBox::clicked, this, &SettingsView::setReopenTab);

